packages/components/src/TabBar/TabBar.component.js

```diff
@@ -164,41 +164,57 @@ function TabBar(props) {
 				)}
 				ref={tabBarRef}
 			>
-				{items.map(({ icon, badge, ...item }) => (
-					<NavItem
-						className={classnames(theme['tc-tab-bar-item'], 'tc-tab-bar-item', item.className)}
-						{...item}
-						eventKey={item.key}
-						componentClass="button"
-					>
-						<TooltipTrigger
-							label={badge && badge.label ? `${badge.label} ${item.label}` : item.label}
-							tooltipPlacement={tooltipPlacement}
+				{items.map(({ icon, badge, ...item }) => {
+					let StyledTag = Tag;
+					switch (badge?.bsStyle?.toLowerCase()) {
+						case 'danger':
```

Comment:
Any chance to use `Status.component.js` (talend/ui/packages/components/src/Status/Status.component.js) to compute the style and reduce hardcoded strings?

Reply:
Not really an option, `Status` and `Tag` are two different components